deltacat/compute/compactor/model/compaction_session_audit_info.py

```diff
@@ -0,0 +1,658 @@
+
+    def set_delta_discovery_time_in_seconds(
+        self, delta_discovery_time_in_seconds: float
+    ) -> CompactionSessionAuditInfo:
+        self["deltaDiscoveryTimeInSeconds"] = delta_discovery_time_in_seconds
+        return self
+
+    def set_output_file_count(
+        self, output_file_count: float
+    ) -> CompactionSessionAuditInfo:
+        self["outputFileCount"] = output_file_count
+        return self
+
+    def set_output_size_bytes(
+        self, output_size_bytes: float
+    ) -> CompactionSessionAuditInfo:
+        self["outputSizeBytes"] = output_size_bytes
+        return output_size_bytes
+
+    def set_output_size_pyarrow_bytes(
+        self, output_size_pyarrow_bytes: float
+    ) -> CompactionSessionAuditInfo:
+        self["outputSizePyarrowBytes"] = output_size_pyarrow_bytes
+        return output_size_pyarrow_bytes
+
+    def set_total_cluster_memory_bytes(
+        self, total_cluster_memory_bytes: float
+    ) -> CompactionSessionAuditInfo:
+        self["totalClusterMemoryBytes"] = total_cluster_memory_bytes
+        return self
+
+    def set_total_cluster_object_store_memory_bytes(
+        self, total_cluster_object_store_memory_bytes: float
+    ) -> CompactionSessionAuditInfo:
+        self[
+            "totalClusterObjectStoreMemoryBytes"
+        ] = total_cluster_object_store_memory_bytes
+        return self
+
+    def set_untouched_file_count(
+        self, untouched_file_count: int
+    ) -> CompactionSessionAuditInfo:
+        self["untouchedFileCount"] = untouched_file_count
+        return self
+
+    def set_untouched_file_ratio(
+        self, untouched_file_ratio: float
+    ) -> CompactionSessionAuditInfo:
+        self["untouchedFileRatio"] = untouched_file_ratio
+        return self
+
+    def set_untouched_record_count(
+        self, untouched_record_count: int
+    ) -> CompactionSessionAuditInfo:
+        self["untouchedRecordCount"] = untouched_record_count
+        return self
+
+    def set_untouched_size_bytes(
+        self, untouched_size_bytes: float
+    ) -> CompactionSessionAuditInfo:
+        self["untouchedSizeBytes"] = untouched_size_bytes
+        return self
+
+    def set_telemetry_time_in_seconds(
+        self, telemetry_time_in_seconds: float
+    ) -> CompactionSessionAuditInfo:
+        self["telemetryTimeInSeconds"] = telemetry_time_in_seconds
+        return self
+
+    # High level methods to save stats
+    def save_step_stats(
+        self,
+        step_name: str,
+        task_results: Union[
+            List[HashBucketResult], List[DedupeResult], List[MaterializeResult]
+        ],
+        task_results_retrieved_at: float,
+        invoke_time_in_seconds: float,
+        task_time_in_seconds: float,
+    ) -> float:
+        """
+        Saves the stats by calling individual setters and returns the cluster telemetry time.
+        """
+
+        last_hb_task_completed_at = max(
+            hb_result.task_completed_at for hb_result in task_results
```

Comment:
```suggestion
        last_task_completed_at = max(
            result.task_completed_at for result in task_results
```
Since this is now used for more than just hash bucketing...

Reply:
fixed.